### PR TITLE
jv-storage

### PR DESCRIPTION
**Pull request by Ppllupold**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -3,17 +3,81 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private int indexForNexElem = 0;
+    private int size;
+    private K[] keyStorage;
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:9]:
```suggestion
    private K[] keys;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,81 @@
...
+    private V[] valueStorage;
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:10]:
```suggestion
    private V[] values;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,81 @@
...
+
+    public StorageImpl() {
+        size = 0;
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:13]:
```suggestion
```
int type = 0 by default

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,17 +3,81 @@
...
+        keyStorage = (K[]) new Object[MAX_SIZE];
+        valueStorage = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:20]: It's better to replace this checking in the beginning of the metho
```suggestion
if (size == MAX_SIZE) {
                throw new RuntimeException("The storage is full");
            }
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -3,17 +3,81 @@
...
+        return valueStorage[getKeyIndex(key)];
     }
 
     @Override
     public int size() {
+        return this.size;
+    }
+
+    private boolean containsKey(K key) {
+        for (int i = 0; i < size; i++) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:51]: this method almost duplicate getKeyIndex method, remove it (see comments above)

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 6]
```diff
@@ -3,17 +3,81 @@
...
+    public StorageImpl() {
+        size = 0;
+        keyStorage = (K[]) new Object[MAX_SIZE];
+        valueStorage = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+
+        if (containsKey(key)) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:21]:
```suggestion
        int index = getKeyIndex(key);
        if (index != -1) {
            values[index] = value;
        } else {
            keys[size] = key;
            values[size] = value;
            size++;
        }
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 7]
```diff
@@ -3,17 +3,81 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private int indexForNexElem = 0;
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:7]: You don't need this variable as you have size

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 8]
```diff
@@ -3,17 +3,81 @@
...
+            valueStorage[indexForNexElem] = value;
+            indexForNexElem++;
+            size++;
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        if (key == null) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:36]:
```suggestion
        int index = getKeyIndex(key);
        return index == -1 ? null : values[index];
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 9]
```diff
@@ -3,17 +3,81 @@
...
+        }
+        if (!containsKey(key)) {
+            return null;
+        }
+        return valueStorage[getKeyIndex(key)];
     }
 
     @Override
     public int size() {
+        return this.size;
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:47]:
```suggestion
        return size;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 10]
```diff
@@ -3,17 +3,81 @@
...
+        for (int i = 0; i < size; i++) {
+            if (key == null ? keyStorage[i] == null
+                    : keyStorage[i] != null && keyStorage[i].equals(key)) {
+                return i;
+            }
+        }
         return -1;
     }
+
+    private void putWithNullKey(K key, V value) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:70]: You don't use this method so remove it

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Review by kerrrusha — APPROVED** [review]

**Pull request closed without merging**